Makefile

```diff
@@ -142,6 +150,89 @@ calico/felix: bin/calico-felix
+K8S_VERSION=1.5.3
+FELIX_K8S=felix-k8s
+.PHONY: k8s-fv-test run-k8s-apiserver stop-k8s-apiserver run-etcd stop-etcd
+k8s-fv-test: calico/felix run-k8s-apiserver k8sfv/k8sfv.test
+	@-docker rm -f $(FELIX_K8S)
+	sleep 1
+	K8S_IP=`$(GET_CONTAINER_IP) calico-k8s-apiserver` && \
+	docker run --detach --privileged --name=$(FELIX_K8S) \
+	-e FELIX_LOGSEVERITYSCREEN=info \
+	-e FELIX_DATASTORETYPE=kubernetes \
+	-e FELIX_PROMETHEUSMETRICSENABLED=true \
+	-e K8S_API_ENDPOINT=https://$${K8S_IP}:6443 \
+	-e K8S_INSECURE_SKIP_TLS_VERIFY=true \
+	-v $${PWD}:/testcode \
+	calico/felix \
+	/bin/sh -c "for n in 1 2; do calico-felix; done"
+	sleep 1
+	K8S_IP=`$(GET_CONTAINER_IP) calico-k8s-apiserver` && \
+	docker exec $(FELIX_K8S) /testcode/k8sfv/k8sfv.test -ginkgo.v https://$${K8S_IP}:6443
+
+run-k8s-apiserver: stop-k8s-apiserver run-etcd
+	ETCD_IP=`$(GET_CONTAINER_IP) calico-etcd` && \
+	docker run --detach \
+	  --name calico-k8s-apiserver \
+	gcr.io/google_containers/hyperkube-amd64:v$(K8S_VERSION) \
+		  /hyperkube apiserver --etcd-servers=http://$${ETCD_IP}:2379 \
+		  --service-cluster-ip-range=10.101.0.0/16 -v=10
+
+stop-k8s-apiserver: stop-etcd
+	@-docker rm -f calico-k8s-apiserver
+	sleep 2
+
+run-etcd: stop-etcd
+	docker run --detach \
+	--name calico-etcd quay.io/coreos/etcd \
+	etcd \
+	--advertise-client-urls "http://127.0.0.1:2379,http://127.0.0.1:4001" \
+	--listen-client-urls "http://0.0.0.0:2379,http://0.0.0.0:4001"
+
+stop-etcd:
+	@-docker rm -f calico-etcd
+
+PROMETHEUS_DATA_DIR := $$HOME/prometheus-data
+K8SFV_PROMETHEUS_DATA_DIR := $(PROMETHEUS_DATA_DIR)/k8sfv
+
+$(K8SFV_PROMETHEUS_DATA_DIR):
+	mkdir -p $@
+
+.PHONY: run-prometheus run-grafana stop-prometheus stop-grafana
+run-prometheus: stop-prometheus $(K8SFV_PROMETHEUS_DATA_DIR)
+	FELIX_IP=`$(GET_CONTAINER_IP) $(FELIX_K8S)` && \
+	sed "s/__FELIX_IP__/$${FELIX_IP}/" < $(K8SFV_DIR)/prometheus/prometheus.yml.in > $(K8SFV_DIR)/prometheus/prometheus.yml
+	docker run --detach --name k8sfv-prometheus \
+	-v $${PWD}/$(K8SFV_DIR)/prometheus/prometheus.yml:/etc/prometheus.yml \
+	-v $(K8SFV_PROMETHEUS_DATA_DIR):/prometheus \
+	prom/prometheus \
+	-config.file=/etc/prometheus.yml \
+	-storage.local.path=/prometheus
+
+stop-prometheus:
+	@-docker rm -f k8sfv-prometheus
+	sleep 2
+
+run-grafana: stop-grafana run-prometheus
+	docker run --detach --name k8sfv-grafana -p 3000:3000 \
+	-v $${PWD}/$(K8SFV_DIR)/grafana:/etc/grafana \
+	-v $${PWD}/$(K8SFV_DIR)/grafana-dashboards:/etc/grafana-dashboards \
+	grafana/grafana --config /etc/grafana/grafana.ini
+	# Wait for it to get going.
+	sleep 5
+	# Configure prometheus data source.
+	PROMETHEUS_IP=`$(GET_CONTAINER_IP) k8sfv-prometheus` && \
+	sed "s/__PROMETHEUS_IP__/$${PROMETHEUS_IP}/" < $(K8SFV_DIR)/grafana-datasources/my-prom.json.in | \
+	curl 'http://admin:admin@127.0.0.1:3000/api/datasources' -X POST \
+	    -H 'Content-Type: application/json;charset=UTF-8' --data-binary @-
+
+stop-grafana:
+	@-docker rm -f k8sfv-grafana
+	sleep 2
```

Comment:
Paranoia about whether the new container will be able to rebind to the relevant port.